### PR TITLE
mosquitto: add fPIC option + bump dependencies + no generic component name + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/mosquitto/2.x/conanfile.py [hunk 1]
```diff
@@ -155,26 +158,26 @@ def package_info(self):
...
-                self.output.info("Appending PATH env var with : {}".format(bin_path))
-                self.env_info.PATH.append(bin_path)
-                self.cpp_info.components[option].requires = ["openssl::openssl", "libmosquitto"]
+                option_comp_name = "mosquitto_{}".format(option)
+                self.cpp_info.components[option_comp_name].libdirs = []
+                self.cpp_info.components[option_comp_name].include_dirs = []
+                self.cpp_info.components[option_comp_name].requires = ["openssl::openssl", "libmosquitto"]
                 if self.options.cjson:
-                    self.cpp_info.components[option].requires.append("cjson::cjson")
+                    self.cpp_info.components[option_comp_name].requires.append("cjson::cjson")
```
Comment (madebr) [inline recipes/mosquitto/2.x/conanfile.py:183]: What if this is built with: `-o mosquitto:apps=False -o mosquitto:clients=False -o mosquitto:cjson=True`?

Comment (SpaceIm) [reply recipes/mosquitto/2.x/conanfile.py:183]: options.cjson is deleted in configure() if clients if False, avoiding to depend on cjson in requirements()

Comment (SpaceIm) [reply recipes/mosquitto/2.x/conanfile.py:183]: So you can ask: what happen when clients if False, apps is True?

And in the lines above: what happend when ssl option is False?

Comment (madebr) [reply recipes/mosquitto/2.x/conanfile.py:183]: Time to introduce a `_with_cjson`/`_with_ssl` method?
Assuming cmake supports building the library with no apps/clients and no cjson present.

**Review by uilianries — APPROVED** [review]

File: recipes/mosquitto/2.x/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/mosquitto/2.x/conanfile.py:183]: yes, to keep in mind for another PR.

**Pull request merged**